hypothesis-python/tests/cover/test_conjecture_minimizer.py

```diff
@@ -44,3 +45,34 @@ def test_can_sort_bytes_by_reordering():
     start = hbytes([5, 4, 3, 2, 1, 0])
     finish = minimize(start, lambda x: set(x) == set(start), random=Random(0))
     assert finish == hbytes([0, 1, 2, 3, 4, 5])
+
+
+def test_can_sort_bytes_by_reordering_partially():
+    start = hbytes([5, 4, 3, 2, 1, 0])
+    finish = minimize(
+        start, lambda x: set(x) == set(start) and x[0] > x[-1],
+        random=Random(0),
+    )
+    assert finish == hbytes([1, 2, 3, 4, 5, 0])
+
+
+# This test is primarily documenting current behaviour. Improvements could
+# render it obsolete.
```

Comment:
Hm. How about changing the `==` to `<=` in this and the other test? That way they're forwards compatible with the shrinker improving, and this asserts an upper bound that should be discoverable.

Reply:
Good call.